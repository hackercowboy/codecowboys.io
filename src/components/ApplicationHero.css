.application-hero-container {
  padding: 100px 50px 0 50px;
  min-height: 830px; }
  @media (min-width: 768px) {
    .application-hero-container {
      min-height: 600px; } }
  .application-hero-container .application-hero-image {
    margin: 10px;
    text-align: center; }
    .application-hero-container .application-hero-image img {
      max-height: 400px; }
  .application-hero-container .application-hero-teaser {
    margin: 0 20px 50px 20px;
    text-align: center; }
    .application-hero-container .application-hero-teaser h2, .application-hero-container .application-hero-teaser p {
      color: white; }
    .application-hero-container .application-hero-teaser h2 {
      margin-top: 30px;
      font-size: 30px; }
    .application-hero-container .application-hero-teaser p {
      font-size: 18px; }
    @media (min-width: 768px) {
      .application-hero-container .application-hero-teaser h2 {
        font-size: 36px; }
      .application-hero-container .application-hero-teaser p {
        font-size: 24px; }
      .application-hero-container .application-hero-teaser img {
        padding-top: 20px; } }
