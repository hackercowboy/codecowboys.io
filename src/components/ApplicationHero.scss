.application-hero {
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: space-evenly;
  padding: 56px 25px 0;

  .application-hero-image {
    max-height: 30vh;
    width: auto;

    img {
      height: auto;
      width: 100%;
    }
  }

  .application-hero-teaser {
    height: 40vh;

    h1 {
      margin: 0;
      padding: 0;
    }

    a {
      color: #fff;
      text-decoration: none;
    }
  }
}

@media only screen and (min-width : 576px) {
  .application-hero {
    align-items: center;
    display: flex;
    flex-direction: row;

    justify-content: space-evenly;

    .application-hero-image {
      height: auto;
      max-height: 50vh;
      max-width: 40%;
    }
  
    .application-hero-teaser {
      max-width: 700px;
      width: 40%;
    }
  }
}
