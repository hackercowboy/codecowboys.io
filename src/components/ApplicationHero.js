import React from 'react';
import PropTypes from 'prop-types';

import {
  Container,
  Row,
  Col,
} from 'reactstrap';

import './ApplicationHero.scss';

const appStoreButton = require('../assets/images/app_store_button.svg');
const appStoreButtonSoon = require('../assets/images/app_store_button_soon.svg');

const ApplicationHero = ({
  image, title, description, soon, link, appStoreLink,
}) => (
  <Container className="application-hero-container">
    <Row>
      <Col sm="12" md="6">
        <div className="application-hero-image">
          <img src={image} className="img-fluid" alt={title} />
        </div>
      </Col>
      <Col>
        <div className="application-hero-teaser">
          { link ? (<h2><a href={link}>{title}</a></h2>) : (<h2>{title}</h2>)}
          { link ? (<p><a href={link}>{description}</a></p>) : (<p>{description}</p>)}
          {soon ? (<img src={appStoreButtonSoon} alt="App Store" />) : (<a href={appStoreLink}><img src={appStoreButton} alt="App Store" /></a>) }
        </div>
      </Col>
    </Row>
  </Container>
);

ApplicationHero.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  soon: PropTypes.bool,
  link: PropTypes.string,
  appStoreLink: PropTypes.string,
};

ApplicationHero.defaultProps = {
  soon: false,
  link: undefined,
  appStoreLink: undefined,
};

export default ApplicationHero;
