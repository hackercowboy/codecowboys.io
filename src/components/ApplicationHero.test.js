import React from 'react';
import { shallow } from 'enzyme';

import ApplicationHero from './ApplicationHero';

describe('<ApplicationHero/>', () => {
  it('should render', () => {
    const wrapper = shallow(<ApplicationHero
      image="image.png"
      title="title"
      link="/test"
      appStoreLink="http://itunes.apple.com/app/letter/id498506154"
      description="description"
    />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
