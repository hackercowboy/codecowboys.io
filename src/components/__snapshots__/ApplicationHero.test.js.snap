// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApplicationHero/> should render with link 1`] = `
"<div className=\\"application-hero\\">
  <div className=\\"application-hero-image\\">
    <a href=\\"/test\\">
      <Image src=\\"image.webp\\" alt=\\"title\\" height={[undefined]} width={[undefined]} />
    </a>
  </div>
  <div className=\\"application-hero-teaser\\">
    <h1>
      <a href=\\"/test\\">
        title
      </a>
    </h1>
    <p>
      <a href=\\"/test\\">
        description
      </a>
    </p>
    <a href=\\"http://itunes.apple.com/app/letter/id498506154\\">
      <Image src=\\"/images/app_store_button.svg\\" alt=\\"App Store\\" height={40} width={135} />
    </a>
  </div>
</div>"
`;

exports[`<ApplicationHero/> should render withou link 1`] = `
"<div className=\\"application-hero\\">
  <div className=\\"application-hero-image\\">
    <Image src=\\"image.webp\\" alt=\\"title\\" height={[undefined]} width={[undefined]} />
  </div>
  <div className=\\"application-hero-teaser\\">
    <h1>
      title
    </h1>
    <p>
      description
    </p>
    <Image src=\\"/images/app_store_button_soon.svg\\" alt=\\"App Store\\" height={40} width={135} />
  </div>
</div>"
`;
